### ISSUE TITLE
[Bug] Vesktop AppIcon icon not present on GNOME panels.

### ISSUE THREAD

**Issue by ottergauze** [Vencord/Vesktop#575]
### Discord Account

ottergauze

### Operating System

Bazzite (Fedora Linux 40.20240427.0)

### Linux Only ~ Desktop Environment

GNOME 46

### What happens when the bug or crash occurs?

Using the AppImage results in the default application icon being used in GNOME's panels, whether it be stock, or augmented using Dash to Panel / Dash to Dock.

The application's icons show properly in the application launcher, applications for installing AppImages like Gear Lever, and in the process tray.

### What is the expected behaviour?

Vesktop should show the appropriate icon in the applications panel, instead of the default executable icon.

### How do you recreate this bug or crash?

1. Download AppImage for Vesktop 1.5.2 for amd64
2. Execute Vesktop
3. Witness the lack of an appropriate icon.
![image](https://github.com/Vencord/Vesktop/assets/16046068/27818fb1-9bc3-4802-a34c-077ca4695dd0)

### Debug Logs

```
[arRPC > ipc] listening at /run/user/1000/discord-ipc-0
[arRPC > websocket] listening on 6463
[arRPC > process] started
```

### Request Agreement

- [X] I have searched the existing issues and found no similar issue
- [X] I am using the latest Vesktop and Vencord versions
- [X] This issue occurs on an official release (not just the AUR or Nix packages)

**Comment by Cynosphere** [comment]
Tangentally related to [this comment](https://github.com/Vencord/Vesktop/issues/298#issuecomment-2092370949)

> For AppImage there is no way to make [a .desktop file icon patch] work.

All that can be done is hope for [this xdg-toplevel-icon feature to be merged](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/269)

> Some edge cases like AppImage bundles could benefit from this as well

**Comment by AstralPhnx** [comment]
Yeah this is partially related to appimages being crapimages. There is a way to fix it though if you modify the .desktop entry using a program like mainmenu and change the "StartupWMClass" variable accordingly it should pick up the proper icon. Far from ideal but workable. Ideally though you'd want to use the flatpak if you're not running using something compiled for your native package manager

**Comment by Vendicated** [comment]
should be fixed
